src/sentry/api/endpoints/project_codeowners_details.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from django.http import Http404
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import ProjectCodeOwners
+
+from .project_codeowners import ProjectCodeOwnerSerializer
+
+from sentry.api.endpoints.project_ownership import ProjectOwnershipMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectCodeOwnersDetailsEndpoint(ProjectEndpoint, ProjectOwnershipMixin):
+    def convert_args(
+        self, request, organization_slug, project_slug, codeowners_id, *args, **kwargs
+    ):
+        args, kwargs = super().convert_args(
+            request, organization_slug, project_slug, *args, **kwargs
+        )
+        try:
+            kwargs["codeowners"] = ProjectCodeOwners.objects.get(
+                id=codeowners_id,
+            )
+        except ProjectCodeOwners.DoesNotExist:
+            raise Http404
+
+        return (args, kwargs)
+
+    def put(self, request, project, codeowners):
+        """
+        Update a CodeOwners
+        `````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          file belongs to.
```

Comment:
Should this be here?

Reply:
@markstory I think it should because it is a path param for this route.
https://github.com/getsentry/sentry/blob/7df73c982e1f8660ed680bd5b4e41ca18ca5063e/src/sentry/api/urls.py#L1678-L1682

Reply:
I meant the referenced to 'file'

Reply:
ahh I'll replace file with `CODEOWNERS`